interface/views.py

```diff
@@ -74,6 +74,15 @@ def homepage(request):
                    'logout_url': redirect(logout_view).url})
 
 
+@login_required
+@csrf_exempt
```

Comment:
You sure this is ok, or even required? Have a read through https://docs.djangoproject.com/en/2.2/ref/csrf/ - in your case I think you can use a hidden input in the form.